pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,141 @@ $title: Use custom JavaScript in AMP pages
   - fstanis
-description: For web experiences requiring a high amount of customization AMP has created amp-script, a component that allows the use of arbitrary JavaScript on your AMP page without affecting the page's overall performance.
+description: A guide to using amp-script, an AMP component that allows you to write custom JavaScript
 ---
 
-AMP strives to provide a consistently great experience to all users across the web by encouraging the use of high-functioning and seamless components that are ready to go out of the box.
+## Introduction
 
-Some web experiences require a high amount of customization that goes beyond the state binding capabilities of [`amp-bind`](../../../documentation/components/reference/amp-bind.md?format=websites) and the dynamic data retrieval and templating functionality of [`amp-list`](../../../documentation/components/reference/amp-list.md?format=websites) and [`amp-mustache`](../../../documentation/components/reference/amp-mustache.md?format=websites). For those one-off cases, AMP has created [`<amp-script>`](../../../documentation/components/reference/amp-script.md?format=websites), a component that allows the use of arbitrary JavaScript on your AMP page without affecting the page's overall performance.
+`amp-script` lets you write and run your own JavaScript in a way that maintains AMP's performance guarantees. Most AMP components enable common web interactions through their own logic, letting you build your page quickly without writing code or importing third-party libraries. By using `amp-script`, you can embrace custom logic for specific use cases or unique needs without losing AMP's benefits.
 
-# Inserting custom JavaScript
+This guide provides background on this component and best practices for its use.
 
-AMP pages support custom JavaScript through the `<amp-script>` component. The example below demonstrates how to use `amp-script` with a JavaScript file loaded from a URL:
 
-```html
-<!doctype html>
-<html ⚡>
-<head>
-  ...
-  <script async custom-element="amp-script" src="https://cdn.ampproject.org/v0/amp-script-0.1.js"></script>
-<body>  
-  ...
-  <amp-script layout="container" src="https://example.com/myfile.js">
-    <p>Initial content that can be modified from JavaScript</p>
-  </amp-script>
-  ...
-</body>
-</html>
-```
+## Web workers
 
-The `<amp-script>` component registers a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) to run on a separate thread than the main page. The Web Worker is given its own copy of the DOM through `amp-script` use of [Worker DOM](https://github.com/ampproject/worker-dom). This allows the Web Worker to use JavaScript libraries, such as [React](https://reactjs.org/) and [jQuery](https://jquery.com/), without modification.
+AMP's validation rules forbid developers from running JavaScript in a webpage via a `<script>` tag. Excessive JavaScript can make websites slow and unresponsive, and AMP wishes to guard against this.
```

Comment:
I would avoid language like "AMP wishes to guard against this" as it sounds more like a hope than something we are actively doing. I'd rephrase or cut the second sentence.